packages/core/src/languages/solidity/predeploys.ts

```diff
@@ -32,32 +31,24 @@ import {
 } from '../../networks'
 
 export const ensureSphinxAndGnosisSafeDeployed = async (
-  provider: SphinxJsonRpcProvider,
+  provider: SphinxJsonRpcProvider | HardhatEthersProvider,
+  wallet: ethers.Wallet,
+  executionMode: ExecutionMode,
   relayers: string[] = [],
   logger?: Logger
 ) => {
-  if (!(await isLiveNetwork(provider))) {
```

Comment:
Context: `ensureSphinxAndGnosisSafeDeployed` is now included in the simulation because otherwise the user would need to deploy the system contracts before they can preview their deployment

Since this function is included in the simulation, this `isLiveNetwork` call would sometimes return the incorrect result. The proper solution is to use the execution mode.